.github/workflows/SQL2019.yaml

```
@@ -0,0 +1,197 @@
+        shell: cmd
+
+      # /MRCACHEDIRECTORY=%CurrentDir% is where we saved the cabinet files
+      # /SAPWD=%dbPassword% /SECURITYMODE=SQL /TCPENABLED=1 allows connecting using localhost without windows account
+      # /FEATURES=SQLEngine,ADVANCEDANALYTICS,SQL_INST_MR,SQL_INST_MPY are the machine learning services options including R and Python
+      # Note that MLS require the initial release installed and then upgrading to newer versions.
+      - name: Install SQL Server 2019
+        run: | 
+          set CurrentDir=%cd%
+          choco install sql-server-2019 -y -params "'/MRCACHEDIRECTORY=%CurrentDir% /SAPWD=%dbPassword% /SECURITYMODE=SQL /TCPENABLED=1 /INDICATEPROGRESS /ACTION=Install /FEATURES=SQLEngine,ADVANCEDANALYTICS,SQL_INST_MR,SQL_INST_MPY /INSTANCENAME=MSSQLSERVER /IACCEPTSQLSERVERLICENSETERMS /IACCEPTROPENLICENSETERMS /IACCEPTPYTHONLICENSETERMS'"
+        shell: cmd
+      
+      - name: Apply SQL Server 2019 Cumulative Update
+        run: choco install sql-server-2019-cumulative-update -y
+        shell: cmd
+
+      - name: Install sqlcmd
+        run: choco install sqlcmd
+        shell: cmd
+
+      # Saves the chocolately logs which will likely hold the errors needed to debug a failed installation of SQL Server 2019.
+      - name: Save Logs as artifact
+        if: always()
+        uses: actions/upload-artifact@v4
+        with:
+          name: Chocolatey Logs
+          path: C:\ProgramData\chocolatey\logs\chocolatey.log
+          retention-days: 20
+
+      - name: Check Connectivity to SQL Database
+        run: |
+          sqlcmd -S localhost -U SA -P %dbPassword% -d Master -l 5 -Q "SELECT @@VERSION"
+        shell: cmd
+
+      - name: Enable External Scripts
+        run: sqlcmd -S localhost -U SA -P %dbPassword% -Q "EXEC sp_configure  'external scripts enabled', 1;"
+
+      - name: Reconfigure
+        run: sqlcmd -S localhost -U SA -P %dbPassword% -Q "RECONFIGURE WITH OVERRIDE;"  
+
+      - name: Restart the Service
+        run: |
+          sqlcmd -S localhost -U SA -P %dbPassword% -Q "SHUTDOWN WITH NOWAIT"
+          timeout /t 5 /nobreak
+          net start "MSSQLSERVER"
+
+      # Validate that MLS works for R.
+      - name: Execute SPEES for R
+        run: sqlcmd -S localhost -U SA -P %dbPassword% -l 5 -Q "
+          EXEC sp_execute_external_script  @language =N'R',
+          @script=N'
+          OutputDataSet <- InputDataSet;
+          ',
+          @input_data_1 =N'SELECT 1 AS hello'
+          WITH RESULT SETS (([hello] int not null));"
+      
+      # Validate that MLS works for Python.
+      - name: Execute SPEES for Python
+        run: sqlcmd -S localhost -U SA -P %dbPassword% -l 5 -Q "
+          EXEC sp_execute_external_script  @language =N'Python',
+          @script=N'OutputDataSet = InputDataSet;',
+          @input_data_1 =N'SELECT 1 AS hello' WITH RESULT SETS (([hello] int not null));"
+
+      - name: Checkout Branch
+        uses: actions/checkout@v2
```

nit use v4. I have a PR to update your 2022 file: #117 
```suggestion
        uses: actions/checkout@v4
```